include/boost/math/special_functions/hypergeometric_1F1.hpp

```diff
@@ -0,0 +1,207 @@
+#include <boost/math/special_functions/detail/hypergeometric_rational.hpp>
+#include <boost/math/special_functions/detail/hypergeometric_1f1_recurrence.hpp>
+#include <boost/math/special_functions/detail/hypergeometric_pade.hpp>
+#include <boost/math/special_functions/detail/hypergeometric_1f1_bessel.hpp>
+#include <boost/math/special_functions/detail/hypergeometric_1f1_scaled_series.hpp>
+
+namespace boost { namespace math { namespace detail {
+
+   // check when 1f1 series can't decay to polynom
+   template <class T>
+   inline bool check_hypergeometric_1f1_parameters(const T& a, const T& b)
+   {
+      BOOST_MATH_STD_USING
+
+         if ((b <= 0) && (b == floor(b)))
+         {
+            if ((a >= 0) || (a < b) || (a != floor(a)))
+               return false;
+         }
+
+      return true;
+   }
+
+   template <class T, class Policy>
+   inline T hypergeometric_1f1_imp(const T& a, const T& b, const T& z, const Policy& pol)
+   {
+      BOOST_MATH_STD_USING // exp, fabs, sqrt
+
+         static const char* const function = "boost::math::hypergeometric_1f1<%1%,%1%,%1%>(%1%,%1%,%1%)";
+
+      if ((z == 0) || (a == 0))
+         return T(1);
+
+      // undefined result:
+      if (!detail::check_hypergeometric_1f1_parameters(a, b))
+         return policies::raise_domain_error<T>(
+            function,
+            "Function is indeterminate for negative integer b = %1%.",
+            b,
+            pol);
+
+      // other checks:
+      if (a == -1)
+         return 1 - (z / b);
+
+      const T b_minus_a = b - a;
+
+      // 0f0 (exp) case;
+      if (b_minus_a == 0)
+         return exp(z);
+
+      if ((b_minus_a == -1))
+      {
+         // for negative integer a and b is reasonable to use truncated series - polynomial
+         if ((a < 0) && (a == ceil(a)))
+            return detail::hypergeometric_1f1_generic_series(a, b, z, pol);
+
+         return (1 + (z / b)) * exp(z);
+      }
+
+      if ((a == 1) && (b == 2))
+         return (exp(z) - 1) / z;
+
+      // asymptotic expansion
+      // check region
+      if (detail::hypergeometric_1f1_asym_region(a, b, z))
+      {
+         // check for poles in gamma for b
+         if ((b > 0) || (b != floor(b)))
+         {
+            //check for poles in gamma for a
+            if (((a > 0) || (a != floor(a))) && (z > 0))
+               return detail::hypergeometric_1f1_asym_positive_series(a, b, z, pol);
+
+            //check for poles in gamma for b
+            if (((b_minus_a > 0) || (b_minus_a != floor(b_minus_a))) && (z < 0))
+               return detail::hypergeometric_1f1_asym_negative_series(a, b, z, pol);
+         }
+      }
+
+      if ((fabs(a * z / b) < 3.5) && (fabs(z * 100) < fabs(b)))
+         return detail::hypergeometric_1f1_rational(a, b, z, pol);
+
+      if (z < -1)
+      {
+         if (a == 1)
+            return detail::hypergeometric_1f1_pade(b, z, pol);
+
+         // Let's otherwise make z positive (almost always)
```

Comment:
I don't understand why you'd want to make z > 0. If z < 0, the series alternates. But of course there are details that may make this naive understanding incorrect.